### PR TITLE
FINERACT-698:Ability to associate a GUID with every request

### PR DESCRIPTION
**Pull request by lingalamadhukar**

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/commands/domain/GuidMapping.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import javax.persistence.Table;
+
+import org.apache.fineract.infrastructure.core.domain.AbstractPersistableCustom;
+
+@Entity
+@Table(name = "m_guid_mapping")
+public class GuidMapping extends AbstractPersistableCustom<Long> {
+
+    @Column(name = "guid", nullable = false, length = 50)
+    private String guid;
```
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/commands/domain/GuidMapping.java:32]: GUID (or UUID) are normally 128-bit integer numbers, but I guess here you've used the term in a more general sense.

**Review by vorburger — COMMENTED** [review]
@lingalamadhukar I was wondering if perhaps a test could be written for this functionality in this PR?

**Comment by vishwasbabu** [comment]
The Jira ticket at https://jira.apache.org/jira/browse/FINERACT-698 has been updated with a sub-task for writing test cases and updating API docs. Would be a good starter task for prospective gsoc / outreachy students

**Comment by muarachmann** [comment]
Hi @vishwasbabu, @lingalamadhukar i have pulled this and it builds fine. 

I am currently testing the rest end point to associate the guid with every transaction from this PR
It built successfully and i managed to create a client, product, payment type etc for testing the route which is

https://localhost:8443/fineract-provider/api/v1/savingsaccounts/{account_id}/transactions?command=deposit
with parameters as mentioned in the PRs docs shown below

```{
  "transactionDate": "16 October 2019",
  "transactionAmount": "10000",
  "paymentTypeId": 1,
  "locale": "en",
  "guid": "1",
  "dateFormat": "dd MMMM yyyy"
}
```

I get the following 500 error here -  https://pastebin.com/e8exU06K

**Comment by vishwasbabu** [comment]
@muarachmann , the relevant error is

`Caused by: org.apache.openjpa.lib.jdbc.ReportingSQLException: Table 'mifostenant-default.m_guid_mapping' doesn't exist {prepstmnt 353510996 INSERT INTO m_guid_mapping (command_source_id, guid) VALUES (?, ?)} [code=1146, state=42S02]
        at org.apache.openjpa.lib.jdbc.LoggingConnectionDecorator.wrap(LoggingConnectionDecorator.java:218)
        at org.apache.openjpa.lib.jdbc.LoggingConnectionDecorator.wrap(LoggingConnectionDecorator.java:194)
        at org.apache.openjpa.lib.jdbc.LoggingConnectionDecorator.access$1000(LoggingConnectionDecorator.java:58)
        at org.apache.openjpa.lib.jdbc.LoggingConnectionDecorator$LoggingConnection$LoggingPreparedStatement.executeUpdate(LoggingConnectionDecorator.java:1133)
        at org.apache.openjpa.lib.jdbc.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:275)
        at org.apache.openjpa.jdbc.kernel.JDBCStoreManager$CancelPreparedStatement.executeUpdate(JDBCStoreManager.java:1791)
        at org.apache.openjpa.jdbc.kernel.PreparedStatementManagerImpl.executeUpdate(PreparedStatementManagerImpl.java:268)
        at org.apache.openjpa.jdbc.kernel.PreparedStatementManagerImpl.flushAndUpdate(PreparedStatementManagerImpl.java:119)
        ... 106 more`

This table is created in this PR. You would have to ensure the relevant patch has been applied in your database

**Comment by muarachmann** [comment]
Oh i notticed re-runing the migrations thanks

**Comment by vorburger** [comment]
@lingalamadhukar thank you for contributing to Mifos! I'm currently doing a quick run of review on all open PRs, and was just wondering if you would still like to get this very old open one merged? If so, could you rebase it, so that we can see if it actually builds on Travis?   (FYI when this PR was raised, this project did not have automated CI on Travis set up; but meanwhile it does.)

Without hearing back within let's say ... 1 month, I would be inclined to just close this (un-merged).

**Comment by vorburger** [comment]
Let's close this now, just to have less open PRs; you can open a new PR with this when you resolved the feedback and conflicts.

**Comment by awasum** [comment]
@lingalamadhukar this work is becoming relevant now on the Fineract Dev List and here: https://github.com/apache/fineract/pull/1951, Please will you like to engage and try to help make your changes happen?

**Pull request closed without merging**